src/main/java/com/puppycrawl/tools/checkstyle/site/PropertiesMacro.java

```diff
@@ -51,6 +52,13 @@
 @Component(role = Macro.class, hint = "properties")
 public class PropertiesMacro extends AbstractMacro {
 
+    /** Set of properties not inherited from the base token configuration. */
+    private static final Set<String> NON_BASE_TOKEN_PROP = Collections.unmodifiableSet(
```

Comment:
1) Why do we need to duplicate https://github.com/checkstyle/checkstyle/blob/ab7ac57697d3d90aba8096841f4b7b639be940e6/src/test/java/com/puppycrawl/tools/checkstyle/internal/XdocsJavaDocsTest.java#L107 and not just use it?

2) All int arrays are assumed to be tokens. Why aren't we using this instead of creating some custom list, which looks incomplete according to number 1?
https://github.com/checkstyle/checkstyle/blob/ab7ac57697d3d90aba8096841f4b7b639be940e6/src/test/java/com/puppycrawl/tools/checkstyle/internal/XdocsPagesTest.java

Reply:
NON_BASE_TOKEN_PROPERTIES is private, not all values are proven to be required for now.
@rnveach , are you ok to make it public? And reuse it.

I am fine to reuse such method if it is applicable.

Reply:
> not all values are proven to be required for now.

Then this is my 3rd question on why is that? That seems odd that the same list of checks and properties are not fully needed for both areas. It still doesn't take away from my item 2 which would make such a list unnecessary.

> are you ok to make it public? And reuse it.

It is coming from a test class, and macros are "production" code. The complete list would need to be moved somewhere.

Reply:
Ok let's move whole collection to macros as is, and test code will use it from main code